src/Framework/Framework/Compilation/ControlTree/Resolved/ResolvedTypeDescriptor.cs

```diff
@@ -138,10 +138,11 @@ public ITypeDescriptor MakeGenericType(params ITypeDescriptor[] typeArguments)
             return new ResolvedTypeDescriptor(genericType);
         }
 
-        public IEnumerable<ITypeDescriptor> FindGenericImplementations(Type genericType)
+        public IEnumerable<ITypeDescriptor> FindGenericImplementations(ITypeDescriptor genericType)
         {
-            return ReflectionUtils.GetBaseTypesAndInterfaces(Type)
-                .Where(i => i.IsGenericType && i.GetGenericTypeDefinition() == genericType)
+            var generic = (genericType as ResolvedTypeDescriptor)?.Type ?? throw new InvalidOperationException($"Only {nameof(ResolvedTypeDescriptor)} sould be used as a parameter.");
```

Comment:
I think we prefer to use the `ToSystemType` function. In case this method is called in VS Extension on a stray ResolvedTypeDescriptor and the Roslyn-backed TypeDescriptor, it will somewhat work.

```suggestion
            var generic = ResolvedTypeDescriptor.ToSystemType(genericType);
```